src/components/layout/Widget.tsx

```diff
@@ -41,11 +41,15 @@ interface Props {
 
 const Widget: React.FC<Props> = ({ children, className }) => {
   const { networkId } = useWalletConnection()
-  const contractAddress = depositApi.getContractAddress(networkId)
+  const contractAddress = networkId ? depositApi.getContractAddress(networkId) : null
 
   return (
     <Wrapper className={className}>
-      <LinkWrapper type="contract" identifier={contractAddress} label={<small>View verified contract</small>} />
+      <LinkWrapper
+        type="contract"
+        identifier={contractAddress || 'No address detected'}
```

Comment:
In case there's no contract, why render this component at all?

Reply:
:+1: